keps/sig-node/127-user-namespaces/README.md

```diff
@@ -0,0 +1,426 @@
+2 is implemented, pods that use volume types not supported in phase 1, fall into
+the phase 2.
+
+All pods in this mode will use _the same_ mapping, chosen by the kubelet, with a
+length 65535, and mapping the range 0-65534 too. IOW, each pod will have its own user
+namespace, but they will map to _the same_ UIDs/GIDs in the host.
+
+Using the same mapping allows for pods to share files and mitigates all the
+listed vulnerabilities (as the host is protected from the container). It is also
+a trivial next-step to take, given that we have phase 1 implemented: just return
+the same mapping if the pod has other volumes.
+
+#### Phase 3: TBD
+
+#### Unresolved
+
+Here is a list of considerations raised in PRs discussion that hasn't yet
+settle. This list is not exhaustive, we are just trying to put the things that
+we don't want to forget or want to highlight it. Some things that are obvious we
+need to tackle are not listed. Let us know if you think it is important to add
+something else to this list:
+
+- We will start with mappings of 64K. Tim Hockin, however, has expressed
+  concerns. See more info on [this Github discussion](https://github.com/kubernetes/enhancements/pull/3065#discussion_r781676224)
+  SergeyKanzhelev [suggested a nice alternative](https://github.com/kubernetes/enhancements/pull/3065#discussion_r807408134),
+  to limit the number of pods so we guarantee enough spare host UIDs in case we
+  need them for the future. There is no final decision yet on how to handle this.
+
+- Tim suggested we should try to not use the whole UID space. Something to keep
+  in mind for next revisions. More info [here](https://github.com/kubernetes/enhancements/pull/3065#discussion_r797100081)
+
+- While we said that if support is not implemented we will throw a clear error,
+  we have not said if it will be API-rejected at admission time or what. I'd
+  like to see how that code looks like before promising something. Was raised [here](https://github.com/kubernetes/enhancements/pull/3065#discussion_r798730922)
+
+- Tim suggested that we might want to allow the container runtimes to choose
+  the mapping and have different runtimes pick different mappings. While KEP
+  authors disagree on this, we still need to discuss it and settle on something.
+  This was [raised here](https://github.com/kubernetes/enhancements/pull/3065#discussion_r798760382)
+
+- (will clarify later with Tim what he really means here). Tim [raised this](https://github.com/kubernetes/enhancements/pull/3065#discussion_r798766024): do
+  we actually want to ship phase 2 as automatic thing? If we do, any workloads
+  that use it can't ever switch to whatever we do in phase 3
+
+### Summary of the Proposed Changes
+
+[This table](https://docs.google.com/presentation/d/1z4oiZ7v4DjWpZQI2kbFbI8Q6botFaA07KJYaKA-vZpg/edit#slide=id.gfd10976c8b_1_41) gives you a quick overview of each phase (note it is outdated, but still useful for a general overview).
```

Comment:
I think that table is very useful and maybe worth porting over here in markdown :)

Reply:
Thanks! I tried to in the past, but give up on markdown formatting. Will do it when switching in the future, whenever we switch this KEP to implementable :)